pkg/pillar/cmd/zedagent/handlecertconfig.go

```diff
@@ -136,3 +141,77 @@ func handleAttestCertDelete(ctxArg interface{}, key string,
 	log.Infof("handleAttestCertDelete for %s", status.Key())
 	return
 }
+
+// Run a task certificate post task, on change trigger
+func certsTask(ctx *zedagentContext, triggerCerts <-chan struct{}) {
+	iteration := 0
+	log.Infoln("starting certs publish task")
+	publishCerts(ctx, iteration)
+
+	// Run a periodic timer so we always update StillRunning
+	stillRunning := time.NewTicker(25 * time.Second)
+	agentlog.StillRunning(agentName+"certs", warningTime, errorTime)
+
+	for {
+		select {
+		case <-triggerCerts:
+			start := time.Now()
+			iteration++
+			publishCerts(ctx, iteration)
+			pubsub.CheckMaxTimeTopic(agentName+"certs", "publishCerts", start,
+				warningTime, errorTime)
+
+		case <-stillRunning.C:
+		}
+		agentlog.StillRunning(agentName+"certs", warningTime, errorTime)
+	}
+}
+
+// prepare the certs list proto message
+func publishCerts(ctx *zedagentContext, iteration int) {
+	var attestReq = &attest.ZAttestReq{}
+
+	attestReq.ReqType = attest.ZAttestReqType_ATTEST_REQ_CERT
+	// no quotes
+
+	// TBD:XXX, get the ECDH Certs here
+
+	log.Debugf("publishCerts to ZedCloud, sending %s", attestReq)
+	sendCertsProtobuf(attestReq, iteration)
+}
+
+// Try all (first free, then rest) until it gets through.
+// Each iteration we try a different port for load spreading.
+// For each port we try all its local IP addresses until we get a success.
+func sendCertsProtobuf(attestReq *attest.ZAttestReq, iteration int) {
+	data, err := proto.Marshal(attestReq)
+	if err != nil {
+		log.Fatal("SendInfoProtobufStr proto marshaling error: ", err)
+	}
+
+	deviceUUID := zcdevUUID.String()
+	zedcloud.RemoveDeferred(deviceUUID)
```

Comment:
This will remove any deferred device info message, since it is the only message which should be identified by the device UUID. 
Note that all other messages which use deferred have their own UUID.
All that is required is to have a unique string on the device, so for the attest message the string can be "attest".

Reply:
sure. have made the change.